src/app/component/general/eco-news/create-news/create-news.component.ts

```diff
@@ -8,11 +11,86 @@ import { preparedImageForCreateEcoNews } from '../../../../links';
 })
 export class CreateNewsComponent implements OnInit {
 
-  link: string = preparedImageForCreateEcoNews;
+  constructor(private router: Router, 
+              private fb: FormBuilder, 
+              private createEcoNewsService: CreateEcoNewsService) {}
 
-  constructor() { }
+  public createNewsForm = this.fb.group({
+    title: ['', [Validators.maxLength(170), Validators.required]],
+    source: [''],
+    content: ['', [Validators.minLength(20), Validators.required]],
+    tags: this.fb.array([])
+  });
+
+  public filters: Array<any> = [
+    {name: 'News', isActive: false},
+    {name: 'Events', isActive: false}, 
+    {name: 'Courses', isActive: false}, 
+    {name: 'Initiatives', isActive: false}, 
+    {name: 'Ads', isActive: false}
+  ];
+
+  public activeLanguage: string = 'en';
+  public link: string;
+  private preparedLink: string = preparedImageForCreateEcoNews;
+  private date: Date = new Date();
+
+  public languages: Array<any> = [
+    {name: 'Ukrainian', lang: 'ua'},
+    {name: 'English', lang: 'en'},
+    {name: 'Russian', lang: 'ru'},
+  ];
 
   ngOnInit() {
+    this.link = this.preparedLink;
+    this.onChanges();
+  }
+
+  private navigateByUrl(): void {
+    this.router.navigateByUrl('/news');
+  }
+
+  public onChanges(): void {
```

Comment:
Please rename this method 'cause it's not clear

Reply:
fixed